src/dependency/Dependency/ServiceProviderExtensions/ServiceProviderExtensions.cs

```diff
@@ -0,0 +1,22 @@
+#nullable enable
+
+using System;
+
+namespace PrimeFuncPack
+{
+    public static partial class ServiceProviderExtensions
+    {
+        private static InvalidCastException CreateUnexpectedServiceTypeException(
```

Comment:
I believe InvalidOperationException should be used here because we do not try to cast. But just some unexpected (invalid) state occurs.